### ISSUE TITLE
[error] PHP Notice:  Undefined index: Sink

### ISSUE THREAD

**Issue by laplasz** [mkweb/pupnp#3]
Hi There!

[Sun May 11 21:34:22 2014] [error] PHP Notice:  Undefined index: Sink in /var/www/pupnp/src/at/mkweb/upnp/backend/UPnP.php on line 118

I think the problem is that the server come back with error for your request - so that should be handled

here is the logs:
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Endpoint: http://192.168.100.100:49154/upnp/control/cds
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Header:
HOST: 192.168.100.100:49154
Content-LENGTH: 433
CONTENT-TYPE: text/xml;charset="utf-8"
USER-AGENT: Linux/2.6.31-1.0 UPnP/1.0 pupnp/0.1
SOAPACTION: "urn:schemas-upnp-org:service:ContentDirectory:1:1#Browse"

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Request:
<?xml version="1.0"?><s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:Browse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1"><ObjectID>0</ObjectID><BrowseFlag>BrowseDirectChildren</BrowseFlag><Filter></Filter><StartingIndex>0</StartingIndex><RequestedCount>0</RequestedCount><SortCriteria></SortCriteria>/u:Browse/s:Body/s:Envelope

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Response:
HTTP/1.1 500 Internal Server Error
CONTENT-LENGTH: 413
CONTENT-TYPE: text/xml; charset=UTF-8
DATE: Sun, 11 May 2014 19:49:20 GMT
EXT:
SERVER: Linux/3.10.25+, UPnP/1.0, MediaTomb/0.12.1
CONNECTION: close

<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
<s:Body>
<s:Fault>
<faultcode>s:Client</faultcode>
<faultstring>UPnPError</faultstring>
<detail>
<UPnPError xmlns="urn:schemas-upnp-org:control-1-0">
<errorCode>-111</errorCode>
<errorDescription>Invalid Action</errorDescription>
</UPnPError>
</detail>
/s:Fault
/s:Body
/s:Envelope

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] ResponseCode: 500
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] HTTP-Code 500 - Create error response
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Return: Array
(
    [detail] => Array
        (
            [errorCode] => -111
            [errorDescription] => Invalid Action
        )

)

**Comment by mkweb** [comment]
Hi Balsz,
thanks for the Logs. Unfortunately there's something missing, I think you've copied them direclty from your browser output?
The browser thinks the containing XML code is HTML and tries to interpret it so it's missing.

Could you please try it again and copy the output from your browser source (ctrl + u)?

I haven't actively worked at this project since a while so I have to work myself in a bit and will try to find your issue in the next days.

Thanks again and best regards from austria.
Mario
-----Ursprüngliche Nachricht-----
Von: laplasz notifications@github.com
Gesendet: Son 11 Mai 2014 22:07
An: mkweb/pupnp pupnp@noreply.github.com
Betreff: [pupnp] [error] PHP Notice: Undefined index: Sink (#3)

Hi There!

[Sun May 11 21:34:22 2014] [error] PHP Notice: Undefined index: Sink in /var/www/pupnp/src/at/mkweb/upnp/backend/UPnP.php on line 118

I think the problem is that the server come back with error for your request - so that should be handled

here is the logs:
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Endpoint: http://192.168.100.100:49154/upnp/control/cds http://192.168.100.100:49154/upnp/control/cds 
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Header:
HOST: 192.168.100.100:49154
Content-LENGTH: 433
CONTENT-TYPE: text/xml;charset="utf-8"
USER-AGENT: Linux/2.6.31-1.0 UPnP/1.0 pupnp/0.1
SOAPACTION: "urn:schemas-upnp-org:service:ContentDirectory:1:1#Browse"

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Request:
<?xml version="1.0"?>0BrowseDirectChildren00/u:Browse/s:Body/s:Envelope

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Response:
HTTP/1.1 500 Internal Server Error
CONTENT-LENGTH: 413
CONTENT-TYPE: text/xml; charset=UTF-8
DATE: Sun, 11 May 2014 19:49:20 GMT
EXT:
SERVER: Linux/3.10.25+, UPnP/1.0, MediaTomb/0.12.1
CONNECTION: close

s:Client
UPnPError

-111
Invalid Action

/s:Fault
/s:Body
/s:Envelope

2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] ResponseCode: 500
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] HTTP-Code 500 - Create error response
2014-05-11 21:49:20 - 192.168.100.102 - [DEBUG] Return: Array
(
 [detail] => Array
 (
 [errorCode] => -111
 [errorDescription] => Invalid Action
 )

)

**Comment by laplasz** [comment]
I think the problem is that response does not contain Sink, or Source tag. Instead of contains notinng since there is no xml at all in the 500 server error response.
What do you think?

**Comment by mkweb** [comment]
> I think the problem is that the server come back with error for your request - so that should be handled

You're absolutely right, there really no error handling and the script tries to find all elements as there were no error occurd. I'll add some and check the other sources.
